### ISSUE TITLE
Pre-commit hooks are executed only when launched from the git root

### ISSUE THREAD

**Issue by savchenko** [gitui-org/gitui#2281]
**Describe the bug**
Pre-commit hook is ignored.

**To Reproduce**
1. `mkdir .githooks && git config --local core.hooksPath .githooks`
2. Create `.githooks/pre-commit` with `echo "Hook triggered" && exit 1` and `chmod +x` it.
3. `git commit` now fails, as expected.
4. `gitui` ignores the hook.

**Expected behavior**
`pre-commit` behaviour is the same when using `git` CLI.

**Screenshots**
N/A.

**Context (please complete the following information):**
 - OS/Distro + Version: `Debian 12`
 - GitUI Version: `gitui 0.25.2`
 - Rust version: `rustc 1.76.0`

**Additional context**
Relates to:
- https://github.com/extrawurst/gitui/pull/1054
- (maybe) https://github.com/extrawurst/gitui/issues/1711

**Comment by extrawurst** [comment]
It seems your repro steps are not correct:
you say you create `mkdir .githooks` but then you create `./githooks/pre-commit` (which is a subtle but very different folder name). if I fix this and create the pre-commit hook inside the path you configure: `.githooks/pre-commit` then gitui correct blocks the commit.

why in your case git rejects and gitui does not is beyond me

<img width="734" alt="Screenshot 2024-06-28 at 09 12 51" src="https://github.com/extrawurst/gitui/assets/776816/50c15911-2429-430d-bf7e-eaac4566ae8e">

**Comment by wessamfathi** [comment]
I'd like to work on this if no one already is.

**Comment by wessamfathi** [comment]
I cannot reproduce this issue on my end, as gitui runs the `pre-commit` correctly whether I'm running it from repo root or sub folders. I'm testing on the `gitui` repo, and from `gitui/src` and `gitui/src/keys`, and it works correctly in all cases.

Using the nightly 2025-01-14 build, macOS Sequoia, rustc 1.85.0.

Is this still an issue for you @savchenko @saiabishek1? Am I following the reproduction steps correctly?

**Comment by extrawurst** [comment]
@wessamfathi could you add a unit test for this in any case - we already have a few for hooks, should be easy to cover (in case we don’t have one maybe already for this subfolder thing)

**Comment by wessamfathi** [comment]
I checked the hook test and it appears that there is one that does that job already `test_hooks_commit_msg_reject_in_subfolder`: it creates a folder `foo` and attempts to commit from within.

Let me know if you mean something else and I'll do it. I'll try to test on Windows as well, but I cannot get OpenSSL to build no matter what I do.

**Comment by extrawurst** [comment]
Yeah I felt we had made a test for this already in the past. If you could get it tested on windows that would be awesome

**Comment by wessamfathi** [comment]
The test passes on Windows 👍

**Comment by extrawurst** [comment]
@savchenko ok will close it as non-reproducible for now. feel free to reopen if this is still a problem on the most recent release

**Comment by naseschwarz** [comment]
Hi @savchenko: I've just tried to reproduce this using a fresh debian 13 container from docker.io, gitui 0.27 downloaded from the releases and the steps to reproduce from your initial issue. My commit is rejected as expected:

![Image](https://github.com/user-attachments/assets/2c818313-0e0b-47d7-be9c-7932929f7d94)

Could you potentially create a syscall trace for us?

```
strace -fo gitui.strace.log gitui
```

... then try to commit in your test repo and upload `gitui.strace.log`.

In case you don't want to publish that such a long log, I suggest you create a private repository, upload it there and invite me.

**Comment by naseschwarz** [comment]
@wessamfathi, do you still want to take care of this? I don't want to do a hostile takeover here. ;)

However, I've put together a failing test: https://github.com/naseschwarz/gitui/tree/issue-2281-reproduce

The failure is due to a combination of *two conditions*, which was not obvious based on the original steps to reproduce, but is now:
1. `core.hooksPath` is set to a relative path
2. gitui is launched with a working directory that is not the working tree root.

**Comment by wessamfathi** [comment]
@naseschwarz oh no worries! Feel free to take it. You're already on track, and now it feels a little bit over my head :)

I'll find something else for sure.
